src/main/java/ognl/OgnlException.java

```diff
@@ -84,15 +68,7 @@ public OgnlException( String msg )
      */
     public OgnlException( String msg, Throwable reason )
     {
-        super( msg );
-        this._reason = reason;
-
-        if (_initCause != null)
-        {
-            try {
-                _initCause.invoke(this, new Object[] { reason });
-            } catch (Exception t) { /** ignore */ }
-        }
+        super( msg , reason, true, false);
```

Comment:
Shouldn't this be controllable from outside?

Reply:
Well, that's was the point of my question about the way to control this behavior change using whatever configuration mechanism is available.
Sorry that I was not clear enough. :-/

Personally, I would add a java prop and initialize a static boolean in `OgnlContext` (mimicking what's already done) to allow the user to control whether the stacktraces are collected when throwing `OgnlException`s or not.
This would allow to cope for unexpected situations when the previous behavior is actually required.

However, since `OgnlContext` is not already provided in any of the `OgnlException`constructor methods, this would stay a JVM-wide setting (meaning you would not be able to control this per OGNL evaluation).
Adding the `OgnlContext` and changing all code paths throwing this exception looks overkill to me.

Thoughts?

Reply:
I meant that you can decide when creating the exception to collect or not the stack trace. I would just add another constructor and expose these `true, false`s and that's all :)

Then we can extend your idea and use `OgnlContext` to control if the exception should be created with stack trace or note (delegate the control out of exception itself).

Reply:
Ah! Indeed, we can add the signature from `java.lang.Exception.Exception(String, Throwable, boolean, boolean)`
No big deal, I'll do it right away.

Reply:
However, I'll leave this new constructor `protected` as this is intended to be used by sub-classes and not by code path throwing the exception - see `java.lang.Throwable.Throwable(String, Throwable, boolean, boolean)`.
Makes sense?

Reply:
Sure, makes sense 👍

Reply:
Done and pushed.